### PR TITLE
Feature/node state ops

### PR DESCRIPTION
**Pull request by ljeub-pometry**
### What changes were proposed in this pull request?

Make lazy node state views support time ops and layer ops

### Why are the changes needed?

so you can get a windowed degree for all nodes in the graph for example

### Does this PR introduce any user-facing change? If yes is this documented?

Adds additional APIs that are documented

### How was this patch tested?

Needs more tests

### CODE REVIEW

File: raphtory/src/db/api/state/ops/node.rs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    type Filtered<G: GraphViewOps<'graph>>: NodeOp<Output = Self::Output>
+        + NodeOpFilter<'graph, Graph = G>
+        + 'graph;
+
+    fn graph(&self) -> &Self::Graph;
+
+    fn filtered<G: GraphViewOps<'graph>>(&self, graph: G) -> Self::Filtered<G>;
+}
+
+#[derive(Debug, Clone, Copy)]
```
Comment (fabianmurariu) [inline raphtory/src/db/api/state/ops/node.rs:36]: I wonder if many of these could just be special cases of a more generic 
``` struct NodeOpFn<O, F: Fn(&GraphStorage, VID) -> O>(F)```
and could be just made from constructors like 
``` impl NodeOpFn {
fn id()-> ...
```

File: raphtory/src/python/graph/node_state.rs
Comment (fabianmurariu) [inline raphtory/src/python/graph/node_state.rs:1]: This file alone makes me SERIOUSLY want to reconsider the way we use generics to filter the graph

File: raphtory/src/db/api/state/ops/node.rs (hunk 1 above)
Comment (ljeub-pometry) [reply raphtory/src/db/api/state/ops/node.rs:36]: If you do this you can no longer name the type which causes a bunch of problems, though maybe we can get away with `fn`

**Review by fabianmurariu — APPROVED** [review]
LGTM

**Pull request merged**